instrumentation/github.com/gocql/gocql/gocql_test.go

```diff
@@ -0,0 +1,479 @@
+			assert.Equal(t, parentSpan.SpanContext().SpanID.String(), span.ParentSpanID.String())
+		default:
+			t.Fatalf("unexpected span name %s", span.Name)
+		}
+		assert.NotNil(t, span.Attributes[cassVersionKey].AsString())
+		assert.Equal(t, cluster.Hosts[0], span.Attributes[cassHostKey].AsString())
+		assert.Equal(t, int32(cluster.Port), span.Attributes[cassPortKey].AsInt32())
+		assert.Equal(t, "up", strings.ToLower(span.Attributes[cassHostStateKey].AsString()))
+	}
+
+	// Check metrics
+	controller.Stop()
+
+	assert.Equal(t, 3, len(exporter.records))
+	expected := []testRecord{
+		{
+			Name:      "cassandra.queries",
+			MeterName: "github.com/gocql/gocql",
+			Labels: []kv.KeyValue{
+				cassHostID("test-id"),
+				cassKeyspace(keyspace),
+				cassStatement(insertStmt),
+			},
+			Number: 1,
+		},
+		{
+			Name:      "cassandra.rows",
+			MeterName: "github.com/gocql/gocql",
+			Labels: []kv.KeyValue{
+				cassHostID("test-id"),
+				cassKeyspace(keyspace),
+			},
+			Number: 0,
+		},
+		{
+			Name:      "cassandra.latency",
+			MeterName: "github.com/gocql/gocql",
+			Labels: []kv.KeyValue{
+				cassHostID("test-id"),
+				cassKeyspace(keyspace),
+			},
+		},
+	}
+
+	for _, record := range exporter.records {
+		name := record.Descriptor().Name()
+		agg := record.Aggregation()
+		switch name {
+		case "cassandra.queries":
+			recordEqual(t, expected[0], record)
+			numberEqual(t, expected[0].Number, agg)
+		case "cassandra.rows":
+			recordEqual(t, expected[1], record)
+			numberEqual(t, expected[1].Number, agg)
+		case "cassandra.latency":
+			recordEqual(t, expected[2], record)
+			// The latency will vary, so just check that it exists
+			if _, ok := agg.(aggregation.MinMaxSumCount); !ok {
+				t.Fatal("missing aggregation in latency record")
+			}
+		default:
+			t.Fatalf("wrong metric %s", name)
+		}
+	}
+
+}
+
+func TestBatch(t *testing.T) {
+	controller := getController(t)
+	defer afterEach()
+	cluster := getCluster()
+	tracer := mocktracer.NewTracer("gocql-test")
+
+	ctx, parentSpan := tracer.Start(context.Background(), "gocql-test")
+
+	session, err := NewSessionWithTracing(
+		ctx,
+		cluster,
+		WithTracer(tracer),
+		WithConnectInstrumentation(false),
+	)
+	assert.NoError(t, err)
+	defer session.Close()
+
+	batch := session.NewBatch(gocql.LoggedBatch).WithContext(ctx)
+	for i := 0; i < 10; i++ {
+		id := gocql.TimeUUID()
+		title := fmt.Sprintf("batch-title-%d", i)
+		stmt := fmt.Sprintf("insert into %s (id, title) values (?, ?)", tableName)
+		batch.Query(stmt, id, title)
+	}
+
+	err = session.ExecuteBatch(batch)
+	assert.NoError(t, err)
+
+	parentSpan.End()
+
+	spans := tracer.EndedSpans()
+	// total spans:
+	// 1 span for the query
+	// 1 span for the local span
+	assert.Equal(t, 2, len(spans))
+	span := spans[0]
+
+	assert.Equal(t, cassBatchQueryName, span.Name)
+	assert.Equal(t, parentSpan.SpanContext().SpanID, span.ParentSpanID)
+	assert.NotNil(t, span.Attributes[cassVersionKey].AsString())
```

Comment:
IIRC `AsString` will return a string which is never `nil`. It might also be more useful for failures to test inclusion here directly.

```suggestion
	assert.Contains(t, span.Attributes, cassVersionKey)
```